### DISCUSSION TITLE
Connection error

### DISCUSSION THREAD

**Discussion by cihangirr** [miguelgrinberg/python-socketio#723] [Q&A]
Hey first of all thanks for that amazing libray. I have little problem with connecting. My client was working well before but after some changes/updates in site i use i got that error . I tried both of the Version compatibility but still getting same result. 
Here is the current version which im using ==>  python-engineio==3.14.2     python-socketio==4.6.0
Where am I missing ? 

![asd1231](https://user-images.githubusercontent.com/68942416/108854123-8c3bca80-75f8-11eb-83e6-36b493d9db8e.png)

**Comment by miguelgrinberg** [comment]
Hard for me to know without logs. Did you enable logs in your server?

**Comment by cihangirr** [comment]
Sir its not my own server, I just made a copy of the website client for botting/automation purposes. I can share a link of the site if there is a allow for it.
Also weird thing is my other bots are works ok on different sites, i can connect to their socketio servers.

**Comment by miguelgrinberg** [comment]
Do you know what version of Socket.IO they use? Try to connect with the latest 5.x version, and also with the 4.6.1 version, which is the latest one that used the older version of the protocol. Other than that I'm not sure how to help, you need to find out from the server what is the reason your connection is not accepted.

**Comment by cihangirr** [comment]
They are using EIO=3, I tried 4x (that should have been suitable) and 5x still same result :(
But i realized something intresting, i can't connect to sites using that ==> "nexus-websocket-a.intercom.io/pubsub" Can it be related with that ?

![tradeşt](https://user-images.githubusercontent.com/68942416/109041212-d9df3280-76df-11eb-8dfb-11e1e1f49891.png)
![pubsu1](https://user-images.githubusercontent.com/68942416/109041280-ea8fa880-76df-11eb-8829-53f736b646fa.png)

**Comment by miguelgrinberg** [comment]
@cihangirr is the nexus site also Socket.IO? It looks like it might be plain WebSocket, which is a completely different thing.

**Comment by cihangirr** [comment]
Yes u are right it seems different thing not socket.io in this case i should have been connected to server.
I really don't see anything wrong, Raw connecting is should have been even work too 

> ![asldkm12313](https://user-images.githubusercontent.com/68942416/109043385-570ba700-76e2-11eb-9038-16d479a03c6d.png)

Do u have any effective suggestion sir ? I appreciate for any suggestion

**Comment by miguelgrinberg** [comment]
As I said, the server is free to reject connections, and that is perfectly fine as far as the protocol goes. You can enable logging in the client if you like to have a more detailed idea of what is the communication between the client and the server, beyond that I don't know what else to suggest outside of asking the maintainers of the server you are connecting to for assistance.

**Comment by cihangirr** [comment]
Alright sir thanks for your time anyway, i will try to do somethings :)

**Comment by cihangirr** [comment]
Sir i believe Connection Error is not relevant with servers rejection it must be something different thing also i oftenly changed the version of the library and got different errors for each changing

I tried connecting by  different library and successfully connected
![adjadk1231](https://user-images.githubusercontent.com/68942416/109248349-ae9b3700-77f6-11eb-9be3-2e2b51c64a41.png)

**Comment by miguelgrinberg** [comment]
That is a websocket library. This is a Socket.IO library. They are not the same thing. If the server is a WebSocket server, then use a websocket client. If the server is a Socket.IO server, then use a Socket.IO client.

**Comment by cihangirr** [comment]
Sir im trying to explain that server is using Socket.IO server not WebSocket server

**Comment by miguelgrinberg** [comment]
You have ignored this:

> You can enable logging in the client if you like to have a more detailed idea of what is the communication between the client and the server

See the troubleshooting section of the docs if you don't know how to do this.

**Comment by cihangirr** [comment]
Sir, already used "troubleshooting" in the first post. I dont know how can i explain that more clearly .

![askdad321](https://user-images.githubusercontent.com/68942416/109309621-9528d800-7854-11eb-9399-e9abdf9b9c40.png)

**Comment by miguelgrinberg** [comment]
@cihangirr I apologize. I look at many issues every day, I cannot keep the context of every one of them in mind.

In any case, I think we are going back in circles. You have logging enabled, there are no error messages, so it's just that the server refused the connection. If you can't ask the people in charge of the server for details on this problem I'm not sure what else can be done.

**Comment by kotx** [comment]
I'm having the same issue. Is it possible to specify the engineio version? The server only seems to work with 3, not 4.

**Comment by miguelgrinberg** [comment]
@kotx The docs have a version compatibility table where you can find what versions to use for the version 3 protocol. https://python-socketio.readthedocs.io/en/latest/intro.html#version-compatibility

**Comment by kotx** [comment]
> @kotx The docs have a version compatibility table where you can find what versions to use for the version 3 protocol. https://python-socketio.readthedocs.io/en/latest/intro.html#version-compatibility

I tried that, though I'm having trouble getting it to work. Python-socketio version 4 seems to request EIO version 4. Version 3 also attempts to request EIO version 4, according to the log. I'm installing versions using `pip install python-socketio[client]==4`. Any pointers?

**Comment by kotx** [comment]
Ah oops, it seems I have to reinstall engineio as well since the version requirements are >=, not ^. It seems to be using EIO 3 now, thanks.

However it seems SID is not persisting in the upgrade to WebSocket.
